lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1399,9 +1410,20 @@ LogicalResult InferenceMapping::mapOperation(Operation *op) {
       })
 
       // Handle the various connect statements that imply a type constraint.
-      .Case<ConnectOp>([&](auto op) { constrainTypes(op.dest(), op.src()); })
-      .Case<PartialConnectOp>(
-          [&](auto op) { partiallyConstrainTypes(op.dest(), op.src()); })
+      .Case<ConnectOp>([&](auto op) {
+        // If the source is an invalid value, we don't set a constraint between
+        // these two types.
+        if (dyn_cast_or_null<InvalidValueOp>(op.src().getDefiningOp()))
+          return;
```

Comment:
I think this may actually be fixing a weird bug where an uninferred thing could be connected to an invalid value with a width. This should be totally impossible to reach from FIRRTL text, though. 👍